librz/bin/format/pe/pe_relocs.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2024 Roee Toledano <roeetoledano10@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include "pe.h"
+#include <rz_util/ht_uu.h>
+
+static unsigned long read_reloc_ent_from_block(RZ_NONNULL RzVector /*<RzBinPeRelocEnt>*/ *relocs, RzBuffer *b, int big_endian, ut32 block_size) {
+	unsigned int bytes_read = sizeof(RzBinPeRelocBlock);
+
+	do {
+		bytes_read += 2;
+		RzBinPeRelocEnt reloc;
+		ut8 ent_buf[sizeof(RzBinPeRelocEnt)];
+		rz_buf_read(b, ent_buf, sizeof(ent_buf));
+		reloc = rz_read_ble16(ent_buf, big_endian);
```

Comment:
maybe use the `rz_buf_read_ble16_offset` which is a special api which updates automatically the offset/address parameter and returns true on successful read.

Reply:
also be sure to check the boolean return value to fail gracefully.